src/core/src/main/java/org/apache/jmeter/samplers/DataStrippingSampleSender.java

```diff
@@ -126,6 +126,24 @@ private static void stripContent(SampleResult result, int level) {
     private static void stripResponse(SampleResult result) {
         result.setBytes(result.getBytesAsLong());
         result.setResponseData(EMPTY_BA);
+
+        if (result.isSuccessful()) {
+            result.setSentBytes(result.getSentBytes());
+            result.setSamplerData(null);
+            result.setRequestHeaders(null);
+
+            result.setBodySize(result.getBodySizeAsLong());
+            result.setDataType(null);
```

Comment:
Do we save much by excluding `dataType`, `dataEncoding`?
What if the server part uses data type, encoding, url, and so in via post-processor?